src/plugins/gpunetio/gpunetio_backend.cpp

```
@@ -16,16 +16,52 @@
+
+bool
+recvAll(int fd, void *buffer, size_t size) {
+    auto *cursor = static_cast<uint8_t *>(buffer);
+    while (size > 0) {
+        const ssize_t received = recv(fd, cursor, size, 0);
+        if (received < 0 && errno == EINTR) {
+            continue;
+        }
+        if (received <= 0) {
+            return false;
+        }
```

Cppcheck (2.21.0)

[style] The function 'supportsRemote' is never used.

(unusedFunction)